platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -184,6 +187,44 @@ class HangingProtocolService {
     this.protocols.set(protocolId, protocol);
   }
 
+  /** Add a given protocol object as active.  Can optionally specify which
+   * implementation of a protocol id is the active one.  This is typically
+   * used to allow different choices for the "default" or "mpr" protocols.
+   */
+  public addActiveProtocol(id: string, protocol?: Protocol): void {
+    if (!id) {
+      throw new Error(`Protocol ${protocol} added without an id`);
+    }
+    console.log('Active protocol set', id, protocol);
+    if (protocol && this.protocols.get(id) !== protocol) {
```

Comment:
Shouldn't this be retrieved from the `this.getProtocolById` so that the dynamic ones can also work?

Reply:
No, the point is one can reconfigure a specific default here - maybe I don't really care about combining that functionality.  Will remove the protocol entirely.